service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfProceedingsController.java

```diff
@@ -104,17 +104,45 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(
             .getNoticeOfProceedingTemplateData(caseData);
 
         List<DocmosisTemplates> templateTypes = getProceedingTemplateTypes(caseData);
-        
+
         List<Document> uploadedDocuments = generateAndUploadDocuments(userId, authorization, templateData,
             templateTypes);
 
-        caseDetails.getData().put("noticeOfProceedingsBundle", createNoticeOfProceedingsCaseData(uploadedDocuments));
+        List<Element> noticeOfProceedingCaseData = createNoticeOfProceedingsCaseData(uploadedDocuments);
+
+        if (callbackRequest.getCaseDetailsBefore() != null
+            && callbackRequest.getCaseDetailsBefore().getData().get("noticeOfProceedingsBundle") != null) {
+            CaseData caseDataBefore = mapper.convertValue(callbackRequest.getCaseDetailsBefore().getData(),
+                CaseData.class);
+
+            noticeOfProceedingCaseData.addAll(getRemovedDocumentBundles(caseDataBefore, templateTypes));
+        }
+
+        caseDetails.getData().put("noticeOfProceedingsBundle", noticeOfProceedingCaseData);
 
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseDetails.getData())
             .build();
     }
 
+    private List<Element> getRemovedDocumentBundles(CaseData caseData, List<DocmosisTemplates> templateTypes) {
+        List<String> templateTypeTitles = templateTypes.stream().map(DocmosisTemplates::getDocumentTitle)
+            .collect(Collectors.toList());
+
+        ImmutableList.Builder<Element> removedDocumentBundles = ImmutableList.builder();
+
+        caseData.getNoticeOfProceedingsBundle().forEach(element -> {
+            String filename = element.getValue().getDocument().getFilename();
+            filename = filename.substring(0, filename.lastIndexOf("."));
```

Comment:
Maybe there is something from FileNameUtils? Like FileNameUtils.getName?

Reply:
fileNameUtils.getBaseName(String) is what we needed :)